docs/3.x upgrade guide.md

```diff
@@ -15,3 +15,25 @@ For a slightly more elaborate setup, [@babel/preset-env](https://babeljs.io/docs
 From `@babel/preset-env`'s docs
 
 > We leverage [`browserslist`, `compat-table`, and `electron-to-chromium`] to maintain mappings of which version of our supported target environments gained support of a JavaScript syntax or browser feature, as well as a mapping of those syntaxes and features to Babel transform plugins and core-js polyfills.
+
+### Typescript usage of withTheme
+
+The typings for `withTheme` have been updated to more accurately reflect its nature as a factory function that returns a `Form` component and to properly type the ref forwarded to this component. See more about [the changes here](https://github.com/rjsf-team/react-jsonschema-form/pull/2279)
+
+If you're currently using `withTheme`, the typing for the formData was always `any`, as the generated Form component was not able to accept a generic argument.
+
+In v3.x these typings will properly infer formData if no type is given, or validate it if one is, similar to how the exported Form from core would work.
```

Comment:
How is a type "given"? Should we show an example code of how to do so?

Reply:
Ping @mattcosta7